client/src/main/java/org/asynchttpclient/cookie/CookieEncoder.java

```diff
@@ -63,8 +63,7 @@ public static String encode(Collection<Cookie> cookies) {
         } else {
             Cookie[] cookiesSorted = cookies.toArray(new Cookie[cookies.size()]);
             Arrays.sort(cookiesSorted, COOKIE_COMPARATOR);
-            for (int i = 0; i < cookiesSorted.length; i++) {
-                Cookie cookie = cookiesSorted[i];
+            for (Cookie cookie : cookiesSorted) {
```

Comment:
@slandelle what iterator do you mean? This is array, `for each` in arrays don't use iterator. Compiler extract this loop into your version, It has nothing to do with JIT. I can show you bytecode if you want :).